src/Bicep.Core/Syntax/SyntaxExtensions.cs

```diff
@@ -90,6 +90,12 @@ public static bool HasSecureDecorator(this DecorableSyntax syntax, IBinder binde
         return SemanticModelHelper.TryGetDecoratorInNamespace(binder, typeManager.GetDeclaredType, syntax, SystemNamespaceType.BuiltInName, LanguageConstants.ParameterSecurePropertyName) is not null;
     }
 
+    /// <summary>
+    /// Checks if the syntax has a decorator with the specified name.
+    /// </summary>
+    public static bool HasDecorator(this DecorableSyntax syntax, string decoratorName)
+        => syntax.Decorators.Any(d => d.Expression is FunctionCallSyntax functionCall && functionCall.NameEquals(decoratorName));
+
```

Comment:
Decorators can be fully qualified, so you can't always make this determination based on syntax alone. There is a helper method that handles instance function calls, but you would need to call it with the model:

https://github.com/Azure/bicep/blob/56292d609041abb01b3909eb00849e66d499f1f5/src/Bicep.Core/Semantics/SemanticModelHelper.cs#L48-L64